makelib/uptest.mk

```diff
@@ -73,37 +73,112 @@ e2e: build controlplane.down controlplane.up $(UPTEST_LOCAL_DEPLOY_TARGET) uptes
 #
 #  Example: `render.crossplane.io/composition-path: "apis/kcl/composition.yaml"`
 #
+# Alternatively the user is able to specify `UPTEST_RENDER_FILES` with a comma or space-separated list
+# of files to render.
+#
+#  Example: `make render UPTEST_RENDER_FILES="path/to/example.yaml,another-example.yaml`
+UPTEST_RENDER_FILES ?=
+UPTEST_EXAMPLES_FOLDER ?= ./examples
 render: $(CROSSPLANE_CLI) ${YQ}
-	@indir="./examples"; \
+	@indir="$(UPTEST_EXAMPLES_FOLDER)"; \
 	rm -rf "$(CACHE_DIR)/render"; \
 	mkdir -p "$(CACHE_DIR)/render" || true; \
-	for file in $$(find $$indir -type f -name '*.yaml' ); do \
+	if [ -z $${UPTEST_RENDER_FILES} ]; then \
+		FILES=$$(find $$indir -type f -name '*.yaml' ); \
+	else \
+		FILES=$$(echo $${UPTEST_RENDER_FILES} | sed 's/,/ /g'); \
+	fi; \
+	$(INFO) Files to render: $${FILES}; \
+	for file in $${FILES}; do \
 		doc_count=$$(${YQ} eval 'documentIndex' $$file | wc -l); \
 		for i in $$(seq 0 $$(($$doc_count - 1))); do \
 			COMPOSITION=$$(${YQ} eval "select(documentIndex == $$i) | .metadata.annotations.\"render.crossplane.io/composition-path\"" $$file); \
 			FUNCTION=$$(${YQ} eval "select(documentIndex == $$i) | .metadata.annotations.\"render.crossplane.io/function-path\"" $$file); \
 			ENVIRONMENT=$$(${YQ} eval "select(documentIndex == $$i) | .metadata.annotations.\"render.crossplane.io/environment-path\"" $$file); \
 			OBSERVE=$$(${YQ} eval "select(documentIndex == $$i) | .metadata.annotations.\"render.crossplane.io/observe-path\"" $$file); \
+			IS_CROSSPLANE_OBJ=$$(${YQ} eval "select(documentIndex == $$i).apiVersion" $$file | grep -E "^pkg\.crossplane\.io|^apiextensions\.crossplane\.io"); \
+			if [ ! -z "$$IS_CROSSPLANE_OBJ" ]; then \
+				continue; \
+			fi; \
 			if [[ "$$ENVIRONMENT" == "null" ]]; then \
 				ENVIRONMENT=""; \
 			fi; \
 			if [[ "$$OBSERVE" == "null" ]]; then \
 				OBSERVE=""; \
 			fi; \
 			if [[ "$$COMPOSITION" == "null" || "$$FUNCTION" == "null" ]]; then \
+				$(WARN) file $$file has document with no annotations for rendering, skipping; \
 				continue; \
 			fi; \
 			OUT_FILE=$$(echo $$file | md5sum | head -c5); \
 			ENVIRONMENT=$${ENVIRONMENT=="null" ? "" : $$ENVIRONMENT}; \
 			OBSERVE=$${OBSERVE=="null" ? "" : $$OBSERVE}; \
+			$(INFO) rendering $$file; \
+			docker ps | grep crossplane-contrib | awk '{ print $$1 }' | xargs -n1 -I% bash -c "docker stop % && docker rm %" 2>&1>/dev/null; \
```

Comment:
did we report the original issue for this one? :)

Reply:
Thanks, good catch! Fix is available now sind crossplane v1.18.0
https://github.com/crossplane/crossplane/pull/5996